src/Guide/Api/Methods.hs

```diff
@@ -221,6 +252,16 @@ dbQuery :: (MonadIO m, EventState event ~ GlobalState, QueryEvent event)
 dbQuery db x = liftIO $
   Acid.query db x
 
+-- Call this whenever any user-made change is applied to the database.
+addEdit :: MonadIO m => DB -> Config -> RequestDetails -> Edit -> m ()
+addEdit db Config{..} RequestDetails{..} edit = unless (isVacuousEdit edit) $ do
+    time <- liftIO getCurrentTime
+    dbUpdate db $ RegisterEdit edit ip time
+    dbUpdate db $ RegisterAction (Action'Edit edit) ip time _baseUrl rdReferer rdUserAgent
+  where
+    ip :: Maybe IP
```

Comment:
Hm, how will it work? `read @(Maybe IP)` is going to parse strings like "Just 251.35.23.186". Maybe you wanted to use `readMaybe`?